src/test/java/com/amplitude/AmplitudeLogTest.java

```diff
@@ -0,0 +1,71 @@
+package com.amplitude;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class AmplitudeLogTest {
+  private ByteArrayOutputStream outContent;
+  private ByteArrayOutputStream errContent;
+  private final PrintStream originalOut = System.out;
+  private final PrintStream originalErr = System.err;
+  private final AmplitudeLog amplitudeLog = new AmplitudeLog();
+
+  @BeforeEach
+  public void setUpStreams() {
+    outContent = new ByteArrayOutputStream();
+    errContent = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(outContent));
+    System.setErr(new PrintStream(errContent));
+  }
+
+  @AfterEach
+  public void restoreStreams() throws IOException {
+    outContent.close();
+    errContent.close();
+    System.setOut(originalOut);
+    System.setErr(originalErr);
+  }
+
+  @ParameterizedTest
+  @MethodSource("logArguments")
+  public void testLog(
+      AmplitudeLog.LogMode logMode,
+      String expectedErrorLog,
+      String expectedWarnLog,
+      String expectedDebugLog) {
```

Comment:
I like the framework here for testing outputs to System.err/out, this makes sense for testing all the modes